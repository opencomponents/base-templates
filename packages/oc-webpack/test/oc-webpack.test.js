jasmine.DEFAULT_TIMEOUT_INTERVAL = 100000;

('use strict');

const path = require('path');
const api = require('../index.js');
const webpackConfigurator = require('../lib/configurator');
const webpackCompiler = require('../lib/compiler');

test('module APIs', () => {
  expect(api).toMatchSnapshot();
});

test('webpack configurator', () => {
  const config = webpackConfigurator({
    stats: 'errors-only',
    dependencies: {},
    publishFileName: 'server.js',
    serverPath: '/path/to/server.js'
  });

  // clean paths
  const target = config.module.rules[0].use;
  target[0].loader = path.relative(__dirname, target[0].loader);
  target[0].options.presets[0][0] = path.relative(
    __dirname,
    target[0].options.presets[0][0]
  );
  target[1].loader = path.relative(__dirname, target[1].loader);
  delete config.logger;

  expect(config).toMatchSnapshot();
});

test('webpack compiler', done => {
  const config = webpackConfigurator({
    stats: 'errors-only',
    dependencies: { lodash: '' },
    publishFileName: 'server.js',
    serverPath: path.join(
      __dirname,
      '../../../mocks/jade-component',
      'server.js'
    )
  });

  webpackCompiler(config, (error, serverContentBundled) => {
    expect(serverContentBundled).toMatchSnapshot();
    done();
  });
});

test('webpack compiler verbose', done => {
  const loggerMock = { log: jest.fn() };
  const config = webpackConfigurator({
    logger: loggerMock,
    stats: 'verbose',
    dependencies: { lodash: '' },
    publishFileName: 'server.js',
    serverPath: path.join(
      __dirname,
      '../../../mocks/jade-component',
      'server.js'
    )
  });

  webpackCompiler(config, (error, serverContentBundled) => {
    const consoleOutput = loggerMock.log.mock.calls[0][0];
    expect(serverContentBundled).toMatchSnapshot();
    expect(consoleOutput).toMatch(/Hash:(.*?)01e93c95dfecf93de280/);
    expect(consoleOutput).toMatch(/Entrypoint(.*?)main(.*?)=(.*?)server.js/);
    expect(consoleOutput).toMatch(/external \"lodash\"/);
    done();
  });
});

test('webpack compiler with fatal error', done => {
  const config = webpackConfigurator({
    stats: 'errors-only',
    dependencies: { lodash: '' },
    publishFileName: 'server.js',
    serverPath: path.join(
      __dirname,
      '../../../mocks/jade-component',
      'server.js'
    )
  });

  config.plugins.push(function() {
    this.plugin('run', (compiler, cb) =>
      cb('This is a fatal compilation error')
    );
  });

  webpackCompiler(config, (error, serverContentBundled) => {
    expect(error).toMatchSnapshot();
    done();
  });
});

test('webpack compiler with soft error', done => {
  const config = webpackConfigurator({
    stats: 'errors-only',
    dependencies: { lodash: '' },
    publishFileName: 'server.js',
    serverPath: path.join(__dirname, 'some/not/valid/path', 'server.js')
  });

  webpackCompiler(config, (error, serverContentBundled) => {
    expect(error).toContain(`Entry module not found: Error: Can't resolve`);
    done();
  });
});

test('webpack compiler with warning', done => {
  const loggerMock = { log: jest.fn() };
  const config = webpackConfigurator({
    logger: loggerMock,
    stats: 'errors-only',
    dependencies: { lodash: '' },
    publishFileName: 'server.js',
    serverPath: path.join(
      __dirname,
      '../../../mocks/jade-component',
      'server.js'
    )
  });

  config.plugins.push(function() {
    this.plugin('run', (compiler, cb) => cb.call(compiler));
    this.plugin('done', stats => stats.compilation.warnings.push('A warning'));
  });

  webpackCompiler(config, (error, serverContentBundled) => {
    expect(loggerMock.log.mock.calls[0][0]).toContain('A warning');
    expect(error).toBe(null);
    done();
  });
});
