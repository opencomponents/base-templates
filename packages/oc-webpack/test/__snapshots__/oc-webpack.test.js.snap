// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module APIs 1`] = `
Object {
  "compiler": [Function],
  "configurator": [Function],
}
`;

exports[`webpack compiler 1`] = `
Object {
  "server.js": "module.exports=function(e){function r(n){if(t[n])return t[n].exports;var u=t[n]={i:n,l:!1,exports:{}};return e[n].call(u.exports,u,u.exports,r),u.l=!0,u.exports}var t={};return r.m=e,r.c=t,r.i=function(e){return e},r.d=function(e,t,n){r.o(e,t)||Object.defineProperty(e,t,{configurable:!1,enumerable:!0,get:n})},r.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return r.d(t,\\"a\\",t),t},r.o=function(e,r){return Object.prototype.hasOwnProperty.call(e,r)},r.p=\\"\\",r(r.s=1)}([function(e,r){e.exports=require(\\"lodash\\")},function(e,r,t){\\"use strict\\";Object.defineProperty(r,\\"__esModule\\",{value:!0}),t.d(r,\\"data\\",function(){return n});var n=(t(0),function(e,r){return r(null,{})})}]);
//# sourceMappingURL=server.js.map",
  "server.js.map": "{\\"version\\":3,\\"sources\\":[\\"server.js\\",\\"webpack/bootstrap 01e93c95dfecf93de280\\",\\"external \\\\\\"lodash\\\\\\"\\"],\\"names\\":[\\"module\\",\\"exports\\",\\"modules\\",\\"__webpack_require__\\",\\"moduleId\\",\\"installedModules\\",\\"i\\",\\"l\\",\\"call\\",\\"m\\",\\"c\\",\\"value\\",\\"d\\",\\"name\\",\\"getter\\",\\"o\\",\\"Object\\",\\"defineProperty\\",\\"configurable\\",\\"enumerable\\",\\"get\\",\\"n\\",\\"__esModule\\",\\"object\\",\\"property\\",\\"prototype\\",\\"hasOwnProperty\\",\\"p\\",\\"s\\",\\"require\\",\\"__webpack_exports__\\",\\"data\\",\\"context\\",\\"callback\\"],\\"mappings\\":\\"AAAAA,OAAOC,QACE,SAAUC,GCGnB,QAAAC,GAAAC,GAGA,GAAAC,EAAAD,GACA,MAAAC,GAAAD,GAAAH,OAGA,IAAAD,GAAAK,EAAAD,IACAE,EAAAF,EACAG,GAAA,EACAN,WAUA,OANAC,GAAAE,GAAAI,KAAAR,EAAAC,QAAAD,IAAAC,QAAAE,GAGAH,EAAAO,GAAA,EAGAP,EAAAC,QAvBA,GAAAI,KA+DA,OAnCAF,GAAAM,EAAAP,EAGAC,EAAAO,EAAAL,EAGAF,EAAAG,EAAA,SAAAK,GAA2C,MAAAA,IAG3CR,EAAAS,EAAA,SAAAX,EAAAY,EAAAC,GACAX,EAAAY,EAAAd,EAAAY,IACAG,OAAAC,eAAAhB,EAAAY,GACAK,cAAA,EACAC,YAAA,EACAC,IAAAN,KAMAX,EAAAkB,EAAA,SAAArB,GACA,GAAAc,GAAAd,KAAAsB,WACA,WAA2B,MAAAtB,GAAA,SAC3B,WAAiC,MAAAA,GAEjC,OADAG,GAAAS,EAAAE,EAAA,IAAAA,GACAA,GAIAX,EAAAY,EAAA,SAAAQ,EAAAC,GAAsD,MAAAR,QAAAS,UAAAC,eAAAlB,KAAAe,EAAAC,IAGtDrB,EAAAwB,EAAA,GAGAxB,IAAAyB,EAAA,KDOM,SAAU5B,EAAQC,GEvExBD,EAAAC,QAAA4B,QAAA,WF6EM,SAAU7B,EAAQ8B,EAAqB3B,GAE7C,YACAa,QAAOC,eAAea,EAAqB,cAAgBnB,OAAO,IACnCR,EAAoBS,EAAEkB,EAAqB,OAAQ,WAAa,MAAOC,IACtG,IAEIA,IAFI5B,EAAoB,GAEjB,SAAc6B,EAASC,GAChC,MAAOA,GAAS\\",\\"file\\":\\"server.js\\",\\"sourcesContent\\":[\\"module.exports =\\\\n/******/ (function(modules) { // webpackBootstrap\\\\n/******/ \\\\t// The module cache\\\\n/******/ \\\\tvar installedModules = {};\\\\n/******/\\\\n/******/ \\\\t// The require function\\\\n/******/ \\\\tfunction __webpack_require__(moduleId) {\\\\n/******/\\\\n/******/ \\\\t\\\\t// Check if module is in cache\\\\n/******/ \\\\t\\\\tif(installedModules[moduleId]) {\\\\n/******/ \\\\t\\\\t\\\\treturn installedModules[moduleId].exports;\\\\n/******/ \\\\t\\\\t}\\\\n/******/ \\\\t\\\\t// Create a new module (and put it into the cache)\\\\n/******/ \\\\t\\\\tvar module = installedModules[moduleId] = {\\\\n/******/ \\\\t\\\\t\\\\ti: moduleId,\\\\n/******/ \\\\t\\\\t\\\\tl: false,\\\\n/******/ \\\\t\\\\t\\\\texports: {}\\\\n/******/ \\\\t\\\\t};\\\\n/******/\\\\n/******/ \\\\t\\\\t// Execute the module function\\\\n/******/ \\\\t\\\\tmodules[moduleId].call(module.exports, module, module.exports, __webpack_require__);\\\\n/******/\\\\n/******/ \\\\t\\\\t// Flag the module as loaded\\\\n/******/ \\\\t\\\\tmodule.l = true;\\\\n/******/\\\\n/******/ \\\\t\\\\t// Return the exports of the module\\\\n/******/ \\\\t\\\\treturn module.exports;\\\\n/******/ \\\\t}\\\\n/******/\\\\n/******/\\\\n/******/ \\\\t// expose the modules object (__webpack_modules__)\\\\n/******/ \\\\t__webpack_require__.m = modules;\\\\n/******/\\\\n/******/ \\\\t// expose the module cache\\\\n/******/ \\\\t__webpack_require__.c = installedModules;\\\\n/******/\\\\n/******/ \\\\t// identity function for calling harmony imports with the correct context\\\\n/******/ \\\\t__webpack_require__.i = function(value) { return value; };\\\\n/******/\\\\n/******/ \\\\t// define getter function for harmony exports\\\\n/******/ \\\\t__webpack_require__.d = function(exports, name, getter) {\\\\n/******/ \\\\t\\\\tif(!__webpack_require__.o(exports, name)) {\\\\n/******/ \\\\t\\\\t\\\\tObject.defineProperty(exports, name, {\\\\n/******/ \\\\t\\\\t\\\\t\\\\tconfigurable: false,\\\\n/******/ \\\\t\\\\t\\\\t\\\\tenumerable: true,\\\\n/******/ \\\\t\\\\t\\\\t\\\\tget: getter\\\\n/******/ \\\\t\\\\t\\\\t});\\\\n/******/ \\\\t\\\\t}\\\\n/******/ \\\\t};\\\\n/******/\\\\n/******/ \\\\t// getDefaultExport function for compatibility with non-harmony modules\\\\n/******/ \\\\t__webpack_require__.n = function(module) {\\\\n/******/ \\\\t\\\\tvar getter = module && module.__esModule ?\\\\n/******/ \\\\t\\\\t\\\\tfunction getDefault() { return module['default']; } :\\\\n/******/ \\\\t\\\\t\\\\tfunction getModuleExports() { return module; };\\\\n/******/ \\\\t\\\\t__webpack_require__.d(getter, 'a', getter);\\\\n/******/ \\\\t\\\\treturn getter;\\\\n/******/ \\\\t};\\\\n/******/\\\\n/******/ \\\\t// Object.prototype.hasOwnProperty.call\\\\n/******/ \\\\t__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };\\\\n/******/\\\\n/******/ \\\\t// __webpack_public_path__\\\\n/******/ \\\\t__webpack_require__.p = \\\\\\"\\\\\\";\\\\n/******/\\\\n/******/ \\\\t// Load entry module and return exports\\\\n/******/ \\\\treturn __webpack_require__(__webpack_require__.s = 1);\\\\n/******/ })\\\\n/************************************************************************/\\\\n/******/ ([\\\\n/* 0 */\\\\n/***/ (function(module, exports) {\\\\n\\\\nmodule.exports = require(\\\\\\"lodash\\\\\\");\\\\n\\\\n/***/ }),\\\\n/* 1 */\\\\n/***/ (function(module, __webpack_exports__, __webpack_require__) {\\\\n\\\\n\\\\\\"use strict\\\\\\";\\\\nObject.defineProperty(__webpack_exports__, \\\\\\"__esModule\\\\\\", { value: true });\\\\n/* harmony export (binding) */ __webpack_require__.d(__webpack_exports__, \\\\\\"data\\\\\\", function() { return data; });\\\\nvar _ = __webpack_require__(0);\\\\n\\\\nvar data = function data(context, callback) {\\\\n  return callback(null, {});\\\\n};\\\\n\\\\n/***/ })\\\\n/******/ ]);\\\\n\\\\n\\\\n// WEBPACK FOOTER //\\\\n// server.js\\",\\" \\\\t// The module cache\\\\n \\\\tvar installedModules = {};\\\\n\\\\n \\\\t// The require function\\\\n \\\\tfunction __webpack_require__(moduleId) {\\\\n\\\\n \\\\t\\\\t// Check if module is in cache\\\\n \\\\t\\\\tif(installedModules[moduleId]) {\\\\n \\\\t\\\\t\\\\treturn installedModules[moduleId].exports;\\\\n \\\\t\\\\t}\\\\n \\\\t\\\\t// Create a new module (and put it into the cache)\\\\n \\\\t\\\\tvar module = installedModules[moduleId] = {\\\\n \\\\t\\\\t\\\\ti: moduleId,\\\\n \\\\t\\\\t\\\\tl: false,\\\\n \\\\t\\\\t\\\\texports: {}\\\\n \\\\t\\\\t};\\\\n\\\\n \\\\t\\\\t// Execute the module function\\\\n \\\\t\\\\tmodules[moduleId].call(module.exports, module, module.exports, __webpack_require__);\\\\n\\\\n \\\\t\\\\t// Flag the module as loaded\\\\n \\\\t\\\\tmodule.l = true;\\\\n\\\\n \\\\t\\\\t// Return the exports of the module\\\\n \\\\t\\\\treturn module.exports;\\\\n \\\\t}\\\\n\\\\n\\\\n \\\\t// expose the modules object (__webpack_modules__)\\\\n \\\\t__webpack_require__.m = modules;\\\\n\\\\n \\\\t// expose the module cache\\\\n \\\\t__webpack_require__.c = installedModules;\\\\n\\\\n \\\\t// identity function for calling harmony imports with the correct context\\\\n \\\\t__webpack_require__.i = function(value) { return value; };\\\\n\\\\n \\\\t// define getter function for harmony exports\\\\n \\\\t__webpack_require__.d = function(exports, name, getter) {\\\\n \\\\t\\\\tif(!__webpack_require__.o(exports, name)) {\\\\n \\\\t\\\\t\\\\tObject.defineProperty(exports, name, {\\\\n \\\\t\\\\t\\\\t\\\\tconfigurable: false,\\\\n \\\\t\\\\t\\\\t\\\\tenumerable: true,\\\\n \\\\t\\\\t\\\\t\\\\tget: getter\\\\n \\\\t\\\\t\\\\t});\\\\n \\\\t\\\\t}\\\\n \\\\t};\\\\n\\\\n \\\\t// getDefaultExport function for compatibility with non-harmony modules\\\\n \\\\t__webpack_require__.n = function(module) {\\\\n \\\\t\\\\tvar getter = module && module.__esModule ?\\\\n \\\\t\\\\t\\\\tfunction getDefault() { return module['default']; } :\\\\n \\\\t\\\\t\\\\tfunction getModuleExports() { return module; };\\\\n \\\\t\\\\t__webpack_require__.d(getter, 'a', getter);\\\\n \\\\t\\\\treturn getter;\\\\n \\\\t};\\\\n\\\\n \\\\t// Object.prototype.hasOwnProperty.call\\\\n \\\\t__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };\\\\n\\\\n \\\\t// __webpack_public_path__\\\\n \\\\t__webpack_require__.p = \\\\\\"\\\\\\";\\\\n\\\\n \\\\t// Load entry module and return exports\\\\n \\\\treturn __webpack_require__(__webpack_require__.s = 1);\\\\n\\\\n\\\\n\\\\n// WEBPACK FOOTER //\\\\n// webpack/bootstrap 01e93c95dfecf93de280\\",\\"module.exports = require(\\\\\\"lodash\\\\\\");\\\\n\\\\n\\\\n//////////////////\\\\n// WEBPACK FOOTER\\\\n// external \\\\\\"lodash\\\\\\"\\\\n// module id = 0\\\\n// module chunks = 0\\"],\\"sourceRoot\\":\\"\\"}",
}
`;

exports[`webpack compiler verbose 1`] = `
Object {
  "server.js": "module.exports=function(e){function r(n){if(t[n])return t[n].exports;var u=t[n]={i:n,l:!1,exports:{}};return e[n].call(u.exports,u,u.exports,r),u.l=!0,u.exports}var t={};return r.m=e,r.c=t,r.i=function(e){return e},r.d=function(e,t,n){r.o(e,t)||Object.defineProperty(e,t,{configurable:!1,enumerable:!0,get:n})},r.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return r.d(t,\\"a\\",t),t},r.o=function(e,r){return Object.prototype.hasOwnProperty.call(e,r)},r.p=\\"\\",r(r.s=1)}([function(e,r){e.exports=require(\\"lodash\\")},function(e,r,t){\\"use strict\\";Object.defineProperty(r,\\"__esModule\\",{value:!0}),t.d(r,\\"data\\",function(){return n});var n=(t(0),function(e,r){return r(null,{})})}]);
//# sourceMappingURL=server.js.map",
  "server.js.map": "{\\"version\\":3,\\"sources\\":[\\"server.js\\",\\"webpack/bootstrap 01e93c95dfecf93de280\\",\\"external \\\\\\"lodash\\\\\\"\\"],\\"names\\":[\\"module\\",\\"exports\\",\\"modules\\",\\"__webpack_require__\\",\\"moduleId\\",\\"installedModules\\",\\"i\\",\\"l\\",\\"call\\",\\"m\\",\\"c\\",\\"value\\",\\"d\\",\\"name\\",\\"getter\\",\\"o\\",\\"Object\\",\\"defineProperty\\",\\"configurable\\",\\"enumerable\\",\\"get\\",\\"n\\",\\"__esModule\\",\\"object\\",\\"property\\",\\"prototype\\",\\"hasOwnProperty\\",\\"p\\",\\"s\\",\\"require\\",\\"__webpack_exports__\\",\\"data\\",\\"context\\",\\"callback\\"],\\"mappings\\":\\"AAAAA,OAAOC,QACE,SAAUC,GCGnB,QAAAC,GAAAC,GAGA,GAAAC,EAAAD,GACA,MAAAC,GAAAD,GAAAH,OAGA,IAAAD,GAAAK,EAAAD,IACAE,EAAAF,EACAG,GAAA,EACAN,WAUA,OANAC,GAAAE,GAAAI,KAAAR,EAAAC,QAAAD,IAAAC,QAAAE,GAGAH,EAAAO,GAAA,EAGAP,EAAAC,QAvBA,GAAAI,KA+DA,OAnCAF,GAAAM,EAAAP,EAGAC,EAAAO,EAAAL,EAGAF,EAAAG,EAAA,SAAAK,GAA2C,MAAAA,IAG3CR,EAAAS,EAAA,SAAAX,EAAAY,EAAAC,GACAX,EAAAY,EAAAd,EAAAY,IACAG,OAAAC,eAAAhB,EAAAY,GACAK,cAAA,EACAC,YAAA,EACAC,IAAAN,KAMAX,EAAAkB,EAAA,SAAArB,GACA,GAAAc,GAAAd,KAAAsB,WACA,WAA2B,MAAAtB,GAAA,SAC3B,WAAiC,MAAAA,GAEjC,OADAG,GAAAS,EAAAE,EAAA,IAAAA,GACAA,GAIAX,EAAAY,EAAA,SAAAQ,EAAAC,GAAsD,MAAAR,QAAAS,UAAAC,eAAAlB,KAAAe,EAAAC,IAGtDrB,EAAAwB,EAAA,GAGAxB,IAAAyB,EAAA,KDOM,SAAU5B,EAAQC,GEvExBD,EAAAC,QAAA4B,QAAA,WF6EM,SAAU7B,EAAQ8B,EAAqB3B,GAE7C,YACAa,QAAOC,eAAea,EAAqB,cAAgBnB,OAAO,IACnCR,EAAoBS,EAAEkB,EAAqB,OAAQ,WAAa,MAAOC,IACtG,IAEIA,IAFI5B,EAAoB,GAEjB,SAAc6B,EAASC,GAChC,MAAOA,GAAS\\",\\"file\\":\\"server.js\\",\\"sourcesContent\\":[\\"module.exports =\\\\n/******/ (function(modules) { // webpackBootstrap\\\\n/******/ \\\\t// The module cache\\\\n/******/ \\\\tvar installedModules = {};\\\\n/******/\\\\n/******/ \\\\t// The require function\\\\n/******/ \\\\tfunction __webpack_require__(moduleId) {\\\\n/******/\\\\n/******/ \\\\t\\\\t// Check if module is in cache\\\\n/******/ \\\\t\\\\tif(installedModules[moduleId]) {\\\\n/******/ \\\\t\\\\t\\\\treturn installedModules[moduleId].exports;\\\\n/******/ \\\\t\\\\t}\\\\n/******/ \\\\t\\\\t// Create a new module (and put it into the cache)\\\\n/******/ \\\\t\\\\tvar module = installedModules[moduleId] = {\\\\n/******/ \\\\t\\\\t\\\\ti: moduleId,\\\\n/******/ \\\\t\\\\t\\\\tl: false,\\\\n/******/ \\\\t\\\\t\\\\texports: {}\\\\n/******/ \\\\t\\\\t};\\\\n/******/\\\\n/******/ \\\\t\\\\t// Execute the module function\\\\n/******/ \\\\t\\\\tmodules[moduleId].call(module.exports, module, module.exports, __webpack_require__);\\\\n/******/\\\\n/******/ \\\\t\\\\t// Flag the module as loaded\\\\n/******/ \\\\t\\\\tmodule.l = true;\\\\n/******/\\\\n/******/ \\\\t\\\\t// Return the exports of the module\\\\n/******/ \\\\t\\\\treturn module.exports;\\\\n/******/ \\\\t}\\\\n/******/\\\\n/******/\\\\n/******/ \\\\t// expose the modules object (__webpack_modules__)\\\\n/******/ \\\\t__webpack_require__.m = modules;\\\\n/******/\\\\n/******/ \\\\t// expose the module cache\\\\n/******/ \\\\t__webpack_require__.c = installedModules;\\\\n/******/\\\\n/******/ \\\\t// identity function for calling harmony imports with the correct context\\\\n/******/ \\\\t__webpack_require__.i = function(value) { return value; };\\\\n/******/\\\\n/******/ \\\\t// define getter function for harmony exports\\\\n/******/ \\\\t__webpack_require__.d = function(exports, name, getter) {\\\\n/******/ \\\\t\\\\tif(!__webpack_require__.o(exports, name)) {\\\\n/******/ \\\\t\\\\t\\\\tObject.defineProperty(exports, name, {\\\\n/******/ \\\\t\\\\t\\\\t\\\\tconfigurable: false,\\\\n/******/ \\\\t\\\\t\\\\t\\\\tenumerable: true,\\\\n/******/ \\\\t\\\\t\\\\t\\\\tget: getter\\\\n/******/ \\\\t\\\\t\\\\t});\\\\n/******/ \\\\t\\\\t}\\\\n/******/ \\\\t};\\\\n/******/\\\\n/******/ \\\\t// getDefaultExport function for compatibility with non-harmony modules\\\\n/******/ \\\\t__webpack_require__.n = function(module) {\\\\n/******/ \\\\t\\\\tvar getter = module && module.__esModule ?\\\\n/******/ \\\\t\\\\t\\\\tfunction getDefault() { return module['default']; } :\\\\n/******/ \\\\t\\\\t\\\\tfunction getModuleExports() { return module; };\\\\n/******/ \\\\t\\\\t__webpack_require__.d(getter, 'a', getter);\\\\n/******/ \\\\t\\\\treturn getter;\\\\n/******/ \\\\t};\\\\n/******/\\\\n/******/ \\\\t// Object.prototype.hasOwnProperty.call\\\\n/******/ \\\\t__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };\\\\n/******/\\\\n/******/ \\\\t// __webpack_public_path__\\\\n/******/ \\\\t__webpack_require__.p = \\\\\\"\\\\\\";\\\\n/******/\\\\n/******/ \\\\t// Load entry module and return exports\\\\n/******/ \\\\treturn __webpack_require__(__webpack_require__.s = 1);\\\\n/******/ })\\\\n/************************************************************************/\\\\n/******/ ([\\\\n/* 0 */\\\\n/***/ (function(module, exports) {\\\\n\\\\nmodule.exports = require(\\\\\\"lodash\\\\\\");\\\\n\\\\n/***/ }),\\\\n/* 1 */\\\\n/***/ (function(module, __webpack_exports__, __webpack_require__) {\\\\n\\\\n\\\\\\"use strict\\\\\\";\\\\nObject.defineProperty(__webpack_exports__, \\\\\\"__esModule\\\\\\", { value: true });\\\\n/* harmony export (binding) */ __webpack_require__.d(__webpack_exports__, \\\\\\"data\\\\\\", function() { return data; });\\\\nvar _ = __webpack_require__(0);\\\\n\\\\nvar data = function data(context, callback) {\\\\n  return callback(null, {});\\\\n};\\\\n\\\\n/***/ })\\\\n/******/ ]);\\\\n\\\\n\\\\n// WEBPACK FOOTER //\\\\n// server.js\\",\\" \\\\t// The module cache\\\\n \\\\tvar installedModules = {};\\\\n\\\\n \\\\t// The require function\\\\n \\\\tfunction __webpack_require__(moduleId) {\\\\n\\\\n \\\\t\\\\t// Check if module is in cache\\\\n \\\\t\\\\tif(installedModules[moduleId]) {\\\\n \\\\t\\\\t\\\\treturn installedModules[moduleId].exports;\\\\n \\\\t\\\\t}\\\\n \\\\t\\\\t// Create a new module (and put it into the cache)\\\\n \\\\t\\\\tvar module = installedModules[moduleId] = {\\\\n \\\\t\\\\t\\\\ti: moduleId,\\\\n \\\\t\\\\t\\\\tl: false,\\\\n \\\\t\\\\t\\\\texports: {}\\\\n \\\\t\\\\t};\\\\n\\\\n \\\\t\\\\t// Execute the module function\\\\n \\\\t\\\\tmodules[moduleId].call(module.exports, module, module.exports, __webpack_require__);\\\\n\\\\n \\\\t\\\\t// Flag the module as loaded\\\\n \\\\t\\\\tmodule.l = true;\\\\n\\\\n \\\\t\\\\t// Return the exports of the module\\\\n \\\\t\\\\treturn module.exports;\\\\n \\\\t}\\\\n\\\\n\\\\n \\\\t// expose the modules object (__webpack_modules__)\\\\n \\\\t__webpack_require__.m = modules;\\\\n\\\\n \\\\t// expose the module cache\\\\n \\\\t__webpack_require__.c = installedModules;\\\\n\\\\n \\\\t// identity function for calling harmony imports with the correct context\\\\n \\\\t__webpack_require__.i = function(value) { return value; };\\\\n\\\\n \\\\t// define getter function for harmony exports\\\\n \\\\t__webpack_require__.d = function(exports, name, getter) {\\\\n \\\\t\\\\tif(!__webpack_require__.o(exports, name)) {\\\\n \\\\t\\\\t\\\\tObject.defineProperty(exports, name, {\\\\n \\\\t\\\\t\\\\t\\\\tconfigurable: false,\\\\n \\\\t\\\\t\\\\t\\\\tenumerable: true,\\\\n \\\\t\\\\t\\\\t\\\\tget: getter\\\\n \\\\t\\\\t\\\\t});\\\\n \\\\t\\\\t}\\\\n \\\\t};\\\\n\\\\n \\\\t// getDefaultExport function for compatibility with non-harmony modules\\\\n \\\\t__webpack_require__.n = function(module) {\\\\n \\\\t\\\\tvar getter = module && module.__esModule ?\\\\n \\\\t\\\\t\\\\tfunction getDefault() { return module['default']; } :\\\\n \\\\t\\\\t\\\\tfunction getModuleExports() { return module; };\\\\n \\\\t\\\\t__webpack_require__.d(getter, 'a', getter);\\\\n \\\\t\\\\treturn getter;\\\\n \\\\t};\\\\n\\\\n \\\\t// Object.prototype.hasOwnProperty.call\\\\n \\\\t__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };\\\\n\\\\n \\\\t// __webpack_public_path__\\\\n \\\\t__webpack_require__.p = \\\\\\"\\\\\\";\\\\n\\\\n \\\\t// Load entry module and return exports\\\\n \\\\treturn __webpack_require__(__webpack_require__.s = 1);\\\\n\\\\n\\\\n\\\\n// WEBPACK FOOTER //\\\\n// webpack/bootstrap 01e93c95dfecf93de280\\",\\"module.exports = require(\\\\\\"lodash\\\\\\");\\\\n\\\\n\\\\n//////////////////\\\\n// WEBPACK FOOTER\\\\n// external \\\\\\"lodash\\\\\\"\\\\n// module id = 0\\\\n// module chunks = 0\\"],\\"sourceRoot\\":\\"\\"}",
}
`;

exports[`webpack compiler with fatal error 1`] = `"This is a fatal compilation error"`;

exports[`webpack configurator 1`] = `
Object {
  "devtool": "#source-map",
  "entry": "/path/to/server.js",
  "externals": Array [
    [Function],
    /\\^\\[a-z@\\]\\[a-z\\\\-\\\\/0-9\\\\\\.\\]\\+\\$/i,
  ],
  "module": Object {
    "rules": Array [
      Object {
        "exclude": /node_modules/,
        "test": /\\\\\\.js\\$/,
        "use": Array [
          Object {
            "loader": "../node_modules/babel-loader/lib/index.js",
            "options": Object {
              "cacheDirectory": true,
              "presets": Array [
                Array [
                  "../node_modules/babel-preset-env/lib/index.js",
                  Object {
                    "modules": false,
                    "targets": Object {
                      "node": 4,
                    },
                  },
                ],
              ],
              "retainLines": true,
              "sourceMaps": true,
              "sourceRoot": "/path/to",
            },
          },
          Object {
            "loader": "../node_modules/infinite-loop-loader/index.js",
          },
        ],
      },
    ],
  },
  "output": Object {
    "devtoolFallbackModuleFilenameTemplate": "[absolute-resource-path]?[hash]",
    "devtoolModuleFilenameTemplate": "[absolute-resource-path]",
    "filename": "server.js",
    "libraryTarget": "commonjs2",
    "path": "/path/to/build",
  },
  "plugins": Array [
    UglifyJsPlugin {
      "options": Object {
        "parallel": true,
        "sourceMap": Object {
          "filename": "/path/to/build/server.js",
          "url": "/path/to/build/server.js.map",
        },
      },
    },
    DefinePlugin {
      "definitions": Object {
        "process.env.NODE_ENV": "\\"production\\"",
      },
    },
  ],
  "stats": "errors-only",
  "target": "node",
}
`;
